app/models/card.rb

```diff
@@ -29,6 +29,26 @@ class Card < ActiveRecord::Base
 	has_many :tickets
 	has_one :source_token, as: :tokenizable
 
+
+	# an helpful method for getting the supporter when it's the holder
+	def supporter
+		if holder_type == 'Supporter'
+			holder
+		else
+			nil
+		end
+	end
+
+	# an helpful method for getting the supporter's nonprofit
+	# when the supporter is the holder
+	def supporter_nonprofit
+		if supporter
+			supporter.nonprofit
+		else
+			nil
+		end
```

Comment:
I think this could be just:

```suggestion
		supporter&.nonprofit
```

Reply:
Good catch!